src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1261,63 +1268,84 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             return cachedItem;
         }
 
+        // Trying to find refresh token first, if existed, will try to use the refresh token. 
         Logger.v(TAG, "Checking refresh tokens");
         RefreshItem refreshItem = getRefreshToken(request);
+        AuthenticationResult authResult = null;
         if (!promptUser(request.getPrompt()) && refreshItem != null
                 && !StringExtensions.IsNullOrBlank(refreshItem.mRefreshToken)) {
             Logger.v(TAG, "Refresh token is available and it will attempt to refresh token");
-            return refreshToken(callbackHandle, activity, useDialog, request, refreshItem, true);
-        } else {
-            Logger.v(TAG, "Refresh token is not available");
+            authResult = getTokenWithRefreshToken(callbackHandle, activity, useDialog, request, refreshItem, true);
+            if (authResult != null && !StringExtensions.IsNullOrBlank(authResult.getAccessToken()))
+            {
+                callbackHandle.onSuccess(authResult);
```

Comment:
I agree with you. If there is exception thrown, and it's non-silent request, the callback will potentially be called twice. Thanks for catching it.